controllers/halayerset_controller.go

```diff
@@ -332,8 +329,8 @@ func (r *HALayerSetReconciler) buildHALayerPod(hals *appv1alpha1.HALayerSet, nod
 
 	pod.Spec.HostNetwork = true
 	pod.Spec.HostAliases = []corev1.HostAlias{
-		{IP: hals.Spec.NodesSpec.FirstNodeIP, Hostnames: []string{sno1Name}},
-		{IP: hals.Spec.NodesSpec.SecondNodeIP, Hostnames: []string{sno2Name}},
+		{IP: hals.Spec.NodesSpec.FirstNodeIP, Hostnames: []string{hals.Spec.NodesSpec.FirstNodeName}},
```

Comment:
should we check if `NodeSpec` is nil?

Reply:
IIUC Since `NodesSpec` and it sub-fields are not marked with `omitempty` you will not be able to manipulate the CR to that state.